tests/trainer/test_trainer.py

```diff
@@ -1173,29 +1174,31 @@ def val_dataloader(self):
-        ({"accelerator": "ddp2"}, DDPStrategy, "ddp", "cpu", 0),
-        ({"accelerator": None, "gpus": 1}, SingleDeviceStrategy, "single_device", "gpu", 1),
-        ({"accelerator": "dp", "gpus": 1}, DataParallelStrategy, "dp", "gpu", 1),
-        ({"accelerator": "ddp", "gpus": 1}, DDPStrategy, "ddp", "gpu", 1),
-        ({"accelerator": "ddp_cpu", "num_processes": 2, "gpus": 1}, DDPSpawnStrategy, "ddp_spawn", "cpu", 0),
-        ({"accelerator": "ddp2", "gpus": 1}, DDP2Strategy, "ddp2", "gpu", 1),
-        ({"accelerator": None, "gpus": 2}, DDPSpawnStrategy, "ddp_spawn", "gpu", 2),
-        ({"accelerator": "dp", "gpus": 2}, DataParallelStrategy, "dp", "gpu", 2),
-        ({"accelerator": "ddp", "gpus": 2}, DDPStrategy, "ddp", "gpu", 2),
-        ({"accelerator": "ddp2", "gpus": 2}, DDP2Strategy, "ddp2", "gpu", 2),
-        ({"accelerator": "ddp2", "num_processes": 2}, DDPStrategy, "ddp", "cpu", 0),
-        ({"accelerator": "dp", "num_processes": 2}, DDPStrategy, "ddp", "cpu", 0),
+        ({"accelerator": None}, SingleDeviceStrategy, "single_device", CPUAccelerator, 0),
+        ({"accelerator": "dp"}, DDPStrategy, "ddp", CPUAccelerator, 0),
+        ({"accelerator": "ddp"}, DDPStrategy, "ddp", CPUAccelerator, 0),
+        ({"accelerator": "ddp", "num_processes": 2}, DDPStrategy, "ddp", CPUAccelerator, 0),
+        ({"accelerator": "ddp", "num_nodes": 2}, DDPStrategy, "ddp", CPUAccelerator, 0),
+        ({"accelerator": "ddp_cpu", "num_processes": 2}, DDPSpawnStrategy, "ddp_spawn", CPUAccelerator, 0),
+        ({"accelerator": "ddp2"}, DDPStrategy, "ddp", CPUAccelerator, 0),
+        ({"accelerator": None, "gpus": 1}, SingleDeviceStrategy, "single_device", GPUAccelerator, 1),
+        ({"accelerator": "dp", "gpus": 1}, DataParallelStrategy, "dp", GPUAccelerator, 1),
+        ({"accelerator": "ddp", "gpus": 1}, DDPStrategy, "ddp", GPUAccelerator, 1),
+        ({"accelerator": "ddp_cpu", "num_processes": 2, "gpus": 1}, DDPSpawnStrategy, "ddp_spawn", CPUAccelerator, 0),
+        ({"accelerator": "ddp2", "gpus": 1}, DDP2Strategy, "ddp2", GPUAccelerator, 1),
+        ({"accelerator": None, "gpus": 2}, DDPSpawnStrategy, "ddp_spawn", GPUAccelerator, 2),
+        ({"accelerator": "dp", "gpus": 2}, DataParallelStrategy, "dp", GPUAccelerator, 2),
+        ({"accelerator": "ddp", "gpus": 2}, DDPStrategy, "ddp", GPUAccelerator, 2),
+        ({"accelerator": "ddp2", "gpus": 2}, DDP2Strategy, "ddp2", GPUAccelerator, 2),
+        ({"accelerator": "ddp2", "num_processes": 2}, DDPStrategy, "ddp", CPUAccelerator, 0),
+        ({"accelerator": "dp", "num_processes": 2}, DDPStrategy, "ddp", CPUAccelerator, 0),
     ],
 )
-def test_trainer_config_accelerator(monkeypatch, trainer_kwargs, strategy_cls, strategy_name, _device_type, num_gpus):
+def test_trainer_config_accelerator(
+    monkeypatch, trainer_kwargs, strategy_cls, strategy_name, accelerator_cls, num_gpus
```

Comment:
Same here. There should be a follow-up PR to make the trainer config tests `gpus` agnostic.

Reply:
yeah! I think someone is making some PRs to update these.

Reply:
Feel free to work on this follow-up!